virtcontainers/device/drivers/vfio.go

```diff
@@ -117,23 +103,44 @@ func (device *VFIODevice) DeviceType() config.DeviceType {
 type VFIODevice_Type int32
 
 const (
-        // 0 is reserved for errors.
-        VFIODevice_TYPE_VFIO_NORMAL  VFIODevice_Type = 1
-        VFIODevice_TYPE_VFIO_MDEV    VFIODevice_Type = 2
+        VFIODEVICE_TYPE_VFIO_ERROR    VFIODevice_Type = 0
+        VFIODEVICE_TYPE_VFIO_NORMAL   VFIODevice_Type = 1
+        VFIODEVICE_TYPE_VFIO_MEDIATED VFIODevice_Type = 2
 )
 
-func getVFIODeviceType(deviceSysStr string) (VFIODevice_Type) {
+func getVFIODetails(deviceFileName, iommuDevicesPath string) (deviceBDF, deviceSysfsDev string, vfioDeviceType VFIODevice_Type, err error) {
+
+        tokens := strings.Split(deviceFileName, ":")
+        vfioDeviceType = VFIODEVICE_TYPE_VFIO_ERROR
+        if len(tokens) == 3 {
+                vfioDeviceType = VFIODEVICE_TYPE_VFIO_NORMAL
+        } else {
+                tokens = strings.Split(deviceFileName, "-")
+                if len(tokens) == 5 {
+                        vfioDeviceType = VFIODEVICE_TYPE_VFIO_MEDIATED
+                }
+        }
+
+        err = nil
```

Comment:
You do not need to initialise these. These will be set to nil/empty string.

Reply:
ok, I will modify it in the next review